tests/ios/pages/ios_course_home_tab.py

```diff
@@ -0,0 +1,69 @@
+"""
+    Course Home Page Module
+"""
+
+from tests.ios.pages.ios_base_page import IosBasePage
+from tests.common import values
+from selenium.common.exceptions import NoSuchElementException
+
+
+class IosCourseHomeTab(IosBasePage):
+    """
+    Course Home Tab screen
+    """
+
+    def component_navigation(self, retries=5):
+        """
+        Navigate between components
+
+        Args:
+            retries (int): Maximum number of retries to find the finish button.
+
+        Returns:
+            webdriver element: finish Element
+        """
+        if retries == 0:
+            raise Exception("Finish button not found after maximum retries")
+
+        self.get_next_btn().click()
+
+        if self.get_finish_button():
+            return self.get_finish_button()
```

Comment:
Can we call this method just once?

Reply:
@jawad-khan 
Its the logic, where we need to click on Next button, and check if Finish element appears then return this Finish element, 
that's why we have to call this method inside if.